SETUP.md

```diff
@@ -131,29 +139,131 @@ SPARK_WORKER_OPTS="-Dspark.worker.cleanup.enabled=true, -Dspark.worker.cleanup.a
+>     ```{shell}
+>     sudo apt-get update
+>     sudo apt-get install zip
+>     ```
+
+Once you have confirmed the databricks cluster is *RUNNING*, install the modules within this repository with the following commands:
+
+```{shell}
 cd Recommenders
-zip -r Recommenders.egg .
-```
-* Once your cluster has started, go to the Databricks home workspace, then go to your user and press import.
-* In the next menu there is an option to import a library, it says: `To import a library, such as a jar or egg, click here`. Press click here.
-* Then, at the first drop-down menu, mark the option `Upload Python egg or PyPI`.
-* Then press on `Drop library egg here to upload` and select the the file `Recommenders.egg` you just created.
-* Then press `Create library`. This will upload the zip and make it available in your workspace.
-* Finally, in the next menu, attach the library to your cluster.
-
-To make sure it works, you can now create a new notebook and import the utilities:
+./scripts/databricks_install.sh <CLUSTER_ID>
 ```
+
+</details>
+
+<details>
+<summary><strong><em>Manual setup</em></strong></summary>
+
+To install the repo manually onto Databricks, follow the steps:
+
+1. Clone the Microsoft Recommenders repository to your local computer.
+2. Zip the contents inside the Recommenders folder (Azure Databricks requires compressed folders to have the `.egg` suffix, so we don't use the standard `.zip`):
+
+    ```{shell}
+    cd Recommenders
+    zip -r Recommenders.egg .
+    ```
+3. Once your cluster has started, go to the Databricks workspace, and select the `Home` button.
+4. Your `Home` directory should appear in a panel. Right click within your directory, and select `Import`.
+5. In the pop-up window, there is an option to import a library, where it says: `(To import a library, such as a jar or egg, click here)`. Select `click here`.
+6. In the next screen, select the option `Upload Python Egg or PyPI` in the first menu.
+7. Next, click on the box that contains the text `Drop library egg here to upload` and use the file selector to choose the `Recommenders.egg` file you just created, and select `Open`.
+8. Click on the `Create library`. This will upload the egg and make it available in your workspace.
+9. Finally, in the next menu, attach the library to your cluster.
+
+</details>
+
+### Confirm Installation
+
+After installation, you can now create a new notebook and import the utilities from Databricks in order to confirm that the import worked.
+
+```{python}
 import reco_utils
 ```
 
-### Troubleshooting for Azure Databricks
+### Troubleshooting Installation on Azure Databricks
+
 * For the [reco_utils](reco_utils) import to work on Databricks, it is important to zip the content correctly. The zip has to be performed inside the Recommenders folder, if you zip directly above the Recommenders folder, it won't work.
 
+## Prepare Azure Databricks for Operationalization
```

Comment:
@nikhilrj - please take a look at this SETUP.

Reply:
LGTM!

Reply:
@nikhilrj hmmm good point. But basically the install-script does the same thing as the manual steps... Maybe we can explain what the script does instead of having those contents as 'manual installation'?

Reply:
Just read your comment below. I agree the SETUP is a bit long now... what's other's thought?

Reply:
i think it's long, but i think the benefits of having all setup in 1 place is worth it. I think we should provide manual information in case someone can't run the scripts for some reason.

However, there are a few different ways to do that:
* Say something like see the scripts and comments in the scripts for manual dependencies, and include links to documentation on how to add libraries, etc. The scripts do implement all the steps, so that is in some ways self-documenting
* Another option would be to have a separate SETUP_MANUAL.md file, and use that as a an Appendix of sorts, where we could reference in the default SETUP.md.

I think if we want to clean it up first bullet is probably a good way to do it. I'd say let's go ahead and merge, and have that as a follow-up action.